jerry-core/ecma/operations/ecma-function-object.cpp

```diff
@@ -731,11 +804,45 @@ ecma_op_function_call (ecma_object_t *func_obj_p, /**< Function object */
   {
     JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_BOUND_FUNCTION);
 
-    JERRY_UNIMPLEMENTED ("Bound functions are not implemented.");
+    /* 2-3. */
+    ecma_property_t *bound_this_prop_p;
+    ecma_property_t *target_function_prop_p;
+
+    bound_this_prop_p = ecma_find_internal_property (func_obj_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS);
+    target_function_prop_p = ecma_get_internal_property (func_obj_p,
+                                                         ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_TARGET_FUNCTION);
+
+    ecma_object_t *target_func_obj_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t,
+                                                                  target_function_prop_p->u.internal_property.value);
+
+    ecma_value_t bound_this_value = 0;
```

Comment:
I've checked this, it looks the `bound_this_prop_p` never could be `NULL`.
Seems it remained from a previous version. Thanks for the notice.